openfisca_france/model/prelevements_obligatoires/prelevements_sociaux/cotisations_sociales/exonerations.py

```diff
@@ -13,6 +14,124 @@ class jei_date_demande(Variable):
+    reference = [
+        'https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000037947610/',
+        'https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000038026966'
+        ]
+    definition_period = MONTH
+    set_input = set_input_divide_by_period
+    documentation = '''
+        Exonération de cotisations et contributions employeur sur les bas salaires.
+
+        Non modélisé (2022):
+        La durée maximale d’application de l’exonération TO-DE est fixée à 119 jours
+        consécutifs ou non, par employeur, par salarié et par année civile.
+    '''
+
+    def formula_2019(individu, period, parameters):
+        # l'individu est le travailleur occasionnel
+        eligible = individu('exoneration_cotisations_employeur_tode_eligibilite', period)
+
+        # cotisations assurances sociales agricoles (ASA) - identiques régime général
+        mmid_employeur = individu('mmid_employeur', period)
+        allegement_cotisation_maladie = individu('allegement_cotisation_maladie', period)  # si rémunération <= 2.5 smic
+        cotisations_asa = mmid_employeur + allegement_cotisation_maladie
+
+        # cotisations allocations familiales
+        famille = individu('famille', period)
+
+        # cotisations accident du travail et des maladies professionnelles
+        accident_du_travail = individu('accident_du_travail', period)
+
+        # contribution à l’allocation logement
+        fnal = individu('fnal', period)
+
+        # cotisation vieillesse
+        vieillesse_deplafonnee_employeur = individu('vieillesse_deplafonnee_employeur', period)
+        vieillesse_plafonnee_employeur = individu('vieillesse_plafonnee_employeur', period)
+
+        # contribution à la retraite complémentaire et CEG
+        agirc_arrco_employeur = individu('agirc_arrco_employeur', period)
+        contribution_equilibre_general_employeur = individu('contribution_equilibre_general_employeur', period)
+
+        # contribution de solidarité pour l’autonomie
+        contribution_solidarite_autonomie = individu('contribution_solidarite_autonomie', period)
+
+        # contribution au titre de l’assurance chômage
+        chomage_employeur = individu('chomage_employeur', period)
+
+        assiette_exoneration = (
+            cotisations_asa
+            + famille
+            + accident_du_travail
+            + fnal
+            + vieillesse_deplafonnee_employeur
+            + vieillesse_plafonnee_employeur
+            + agirc_arrco_employeur
+            + contribution_equilibre_general_employeur
+            + contribution_solidarite_autonomie
+            + chomage_employeur
+            )
+
+        # Exonération totale à 1.2 SMIC
+        # Puis dégressive : 1,2 × C/0,40 × (1,6 × montant mensuel du SMIC/ rémunération mensuelle brute hors heures supplémentaires et complémentaires-1)
+        # Devient nulle à 1.6 SMIC
+
+        salaire_de_base = individu('salaire_de_base', period)
```

Comment:
> `salaire_de_base` choisi au regard de la [formule de calcul](https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000038026966) qui indique `rémunération mensuelle brute hors heures supplémentaires et complémentaires`. 
D'un autre côté, `La rémunération mensuelle brute correspond à celle retenue pour le calcul des cotisations de la réduction générale des cotisations patronales (réduction Fillon).`
d'après la [page TO-DE de la MSA](https://www.msa.fr/lfp/employeur/exonerations-travailleurs-occasionnels?p_p_id=com_liferay_journal_content_web_portlet_JournalContentPortlet_INSTANCE_TnUJJSlWXJvY&p_p_lifecycle=0&p_p_state=normal&p_p_mode=view&_com_liferay_journal_content_web_portlet_JournalContentPortlet_INSTANCE_TnUJJSlWXJvY_read_more=2).